hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -1142,12 +1145,12 @@ private void initializeDagger(@NonNull final State state, @NonNull final InitTri
         final var initialStateHash = new InitialStateHash(initialStateHashFuture, roundNum);
 
         final var rosterStore = new ReadableStoreFactory(state).getStore(ReadableRosterStore.class);
+        final var currentRoster = requireNonNull(rosterStore.getActiveRoster());
         final var networkInfo = new StateNetworkInfo(
-                platform.getSelfId().id(),
-                state,
-                requireNonNull(rosterStore.getActiveRoster()),
-                configProvider,
-                () -> requireNonNull(genesisNetworkSupplier).get());
+                platform.getSelfId().id(), state, currentRoster, configProvider, () -> requireNonNull(
+                                genesisNetworkSupplier)
+                        .get());
+        hintsService.initCurrentRoster(currentRoster);
```

Comment:
Why do we need to maintain this in memory data ? Is this not always the same as `rosterState.getActiveRoster`?

Reply:
On `main` we didn't set the current roster until the first call to `reconcileTssState()`; but in this PR we need it available during migration (c.f. `Hedera.onAdoptRoster()` callback).